x/btcstaking/types/btc_slashing_tx.go

```diff
@@ -193,25 +193,28 @@ func (tx *BTCSlashingTx) EncVerifyAdaptorSignature(
 	)
 }
 
-// EncVerifyAdaptorSignatures verifies a list of adaptor signatures, each
+// ParseEncVerifyAdaptorSignatures verifies a list of adaptor signatures, each
 // encrypted by a restaked validator PK and signed by the given PK, w.r.t. the
 // given funding output (in staking or unbonding tx), slashing spend info and
 // slashing tx
-func (tx *BTCSlashingTx) EncVerifyAdaptorSignatures(
+// It returns a list of parsed adaptor signatures in case of successful verification
+func (tx *BTCSlashingTx) ParseEncVerifyAdaptorSignatures(
 	fundingOut *wire.TxOut,
 	slashingSpendInfo *btcstaking.SpendInfo,
 	pk *bbn.BIP340PubKey,
 	valPKs []bbn.BIP340PubKey,
 	sigs [][]byte,
-) error {
-	for i, sig := range sigs {
+) ([]asig.AdaptorSignature, error) {
+	var adaptorSigs []asig.AdaptorSignature = make([]asig.AdaptorSignature, len(sigs))
+	for i, s := range sigs {
+		sig := s
```

Comment:
```suggestion
	for i := range sigs {
		sig := sigs[i]
```
Probably a nitpick but the above seems more natural